tests/index/test_index_spec.lua

```diff
@@ -0,0 +1,197 @@
+-- Tests for lugit2's Index module functions
+
+local fixer = require("Fixtures.fix_repo")
+
+describe(" Index Methods Tests ", function()
+	local lib = require("luagit2")
+	local repo
+	local repo_index
+	local current_directory_path = lfs.currentdir()
+
+	setup(function()
+		lib.luagit2_init()
+	end)
+
+	before_each(function()
+		fixer.set_repo("new_test_repo")
+		repo = lib.luagit2_repository_open("Fixtures/WORKON_REPO/.git")
+		repo_index = lib.luagit2_repository_index(repo)
+	end)
+
+	after_each(function()
+		lib.luagit2_index_free(repo_index)
+		lib.luagit2_repository_free(repo)
+		fixer.set_back()
+	end)
+
+	teardown(function()
+		lib.luagit2_shutdown()
+	end)
+
+	it("Tests Index path",function()
+		local path = lib.luagit2_index_path(repo_index)
+		assert.are.equal(current_directory_path .. "/Fixtures/WORKON_REPO/.git/index" , path)
```

Comment:
@Etiene mam, here I used lfs to check the path. The path string that we get from `luagit2_index_path()` is absolute path. I made it this way so that it runs correctly at any location used.

Reply:
ok!